lib/cache/middleware/base/index.js

```diff
@@ -0,0 +1,277 @@
+'use strict';
+const cacheProvider = require('../../cache'),
+  _ = require('lodash'),
+  q = require('q'),
+  assert = require('assert');
+/**
+ *
+ * @Throws {Error} if is unable to connect with redis.
+ * @param {Object} redis - Redis API configuration.
+ * @param {Number} maxTime - The cache maxTime.
+ * @param {String} prefix - Cache prefix for identification. Default : '' .
+ * @param {String} catalog - Default catalog. Default : null .
+ * @param {Number} catalogDuration - Catalog duration time. Default : -1 (not expire)'' .
+ * {Object} logger - Error logging provider. It must define an `error` function. Default: null
+ * @constructor
+ */
+class Middleware {
```

Comment:
this is not required

Reply:
These are existing old code, removing it.